cranelift/codegen/src/isa/riscv64/inst/mod.rs

```diff
@@ -617,6 +617,12 @@ fn riscv64_get_operands<F: Fn(VReg) -> VReg>(inst: &Inst, collector: &mut Operan
             // gen_prologue is called at emit stage.
             // no need let reg alloc know.
         }
+        &Inst::ElfTlsGetAddr { rd, .. } => {
+            collector.reg_fixed_def(rd, a0());
+            let mut clobbers = Riscv64MachineDeps::get_regs_clobbered_by_call(CallConv::SystemV);
```

Comment:
Could you also add a use for `a0`?

Reply:
@bjorn3 to clarify, what would the use be for? The instruction only has one register arg (`rd`), which is output-only; does something else flow in?

Reply:
Right, the call contained within it uses `a0`, but that one is defined earlier in this macro instruction.

Reply:
Yes, and given that, `a0` is not a use of the pseudoinstruction: it is not live-in, and the register allocator does not need to track a liverange for an input that ends up in `a0`. Purely internal operations can read and write `a0` as many times as they like, since the def constrained to `a0` allows the instruction to have an overall effect of "overwrites `a0` with output value resulting".

So, @yuyang-ok: this code is correct as-is, no worries!

Reply:
Yes ,a0 is aslo used.

Reply:
@cfallin Yes, no need use,`aarch64` did the same.